src/test/scala/latis/input/HapiAdapterSpec.scala

```diff
@@ -0,0 +1,57 @@
+package latis.input
+
+import java.net.URI
+import org.scalatest.FlatSpec
+import org.scalatest.Matchers._
+
+import latis.data._
+import latis.dataset._
+import latis.metadata.Metadata
+import latis.model._
+import latis.ops.Selection
+import latis.util.StreamUtils
+
+object HapiAdapterSpec extends FlatSpec {
+  
+  val model = Function(
+    Scalar(Metadata(
+      "id"   -> "time",
+      "type" -> "string"
+    )),
+    Scalar(Metadata(
+      "id"   -> "tsi_1au",
+      "type" -> "double"
+    ))
+  )
+  
+  val adapter = new HapiCsvAdapter(
+    model,
+    new HapiAdapter.Config(
+      "class" -> "latis.input.HapiCsvAdapter",
+      "id"    -> "sorce_tsi_24hr_l3"
+    )
+  )
+  
+  val baseUri = new URI("http://lasp.colorado.edu/lisird/hapi/")
+  
+  val dataset = new AdaptedDataset(
+    Metadata("sorce_tsi"),
+    model,
+    adapter,
+    baseUri
+  )
+//  val dataset = Dataset.fromName("sorce_tsi") //get from fdml
+  
+  
+  "A hapi csv request with time selections" should "return csv records" in {
+    val ds = dataset.withOperation(Selection("time", ">", "2010-01-01"))
```

Comment:
Note the new `withOperation` that the new dataset stuff supports. The DSL could add a `select` method on `Dataset`.